server/coordinator/procedure/transfer_leader.go

```diff
@@ -0,0 +1,225 @@
+package procedure
+
+import (
+	"context"
+	"github.com/CeresDB/ceresdbproto/pkg/clusterpb"
+	"github.com/CeresDB/ceresmeta/server/cluster"
+	"github.com/CeresDB/ceresmeta/server/schedule"
+	"github.com/looplab/fsm"
+	"github.com/pkg/errors"
+)
+
+const (
+	EventTransferLeaderPrepare = "EventTransferLeaderPrepare"
+	EventTransferLeaderFailed  = "EventTransferLeaderFailed"
+	EventTransferLeaderSuccess = "EventTransferLeaderSuccess"
+
+	StateTransferLeaderBegin   = "StateTransferLeaderBegin"
+	StateTransferLeaderWaiting = "StateTransferLeaderWaiting"
+	StateTransferLeaderFinish  = "StateTransferLeaderFinish"
+	StateTransferLeaderFailed  = "StateTransferLeaderFailed"
+)
+
+var (
+	transferLeaderEvents = fsm.Events{
+		{Name: EventTransferLeaderPrepare, Src: []string{StateTransferLeaderBegin}, Dst: StateTransferLeaderWaiting},
+		{Name: EventTransferLeaderSuccess, Src: []string{StateTransferLeaderWaiting}, Dst: StateTransferLeaderFinish},
+		{Name: EventTransferLeaderFailed, Src: []string{StateTransferLeaderWaiting}, Dst: StateTransferLeaderFailed},
+	}
+	transferLeaderCallbacks = fsm.Callbacks{
+		EventTransferLeaderPrepare: func(event *fsm.Event) {
+			request := event.Args[0].(*CallbackRequest)
+			p := request.p
+			c := request.c
+			handler := request.handler
+			ctx := request.cxt
+			leaderFsm := request.leaderFsm
+			followerFsm := request.followerFsm
+
+			// When `Replication_Factor` == 1, newLeader is nil, find a suitable node and create a newFollower as newLeader
+			if p.newLeader == nil {
+				targetNode, err := c.GetClusterBalancer().SelectNode()
```

Comment:
I think we need to use scheduler to generate `procedure`.
Inside the `procedure`, we don't need to get information from `balancer` .